exec/vm.go

```diff
@@ -0,0 +1,332 @@
+// Copyright 2017 The go-interpreter Authors.  All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// package exec provides functions for executing WebAssembly bytecode.
+package exec
+
+import (
+	"encoding/binary"
+	"errors"
+	"fmt"
+	"math"
+
+	"github.com/go-interpreter/wagon/disasm"
+	"github.com/go-interpreter/wagon/exec/internal/compile"
+	"github.com/go-interpreter/wagon/wasm"
+	ops "github.com/go-interpreter/wagon/wasm/operators"
+)
+
+var MultipleLinearMemoriesErr = errors.New("exec: more than one linear memories in module")
+
+type ErrorInvalidReturnType int8
```

Comment:
Go style suggests an error type should be named `InvalidReturnTypeError`, not `ErrorInvalidReturnType`.

Reference: https://dmitri.shuralyov.com/idiomatic-go#error-variable-naming.

Also, it should be documented or made unexported.